modin/core/storage_formats/pandas/parsers.py

```diff
@@ -564,12 +564,14 @@ def update_row_nums(match):
             has_index_names=is_list_like(header) and len(header) > 1,
             skiprows=skiprows,
             usecols=usecols,
+            skip_blank_lines=False,
             **kwargs,
         )
-        # In excel if you create a row with only a border (no values), this parser will
-        # interpret that as a row of NaN values. pandas discards these values, so we
-        # also must discard these values.
-        pandas_df = parser.read().dropna(how="all")
+        pandas_df = parser.read()
+        for _ in range(pandas_df.shape[0]):
```

Comment:
This loop will be inefficient for even medium-sized datasets. Ideally we will not have the performance of the common cases impacted by the edge cases.

Can we `break` out of this loop in the common case (where there are no blank lines)?